smokey/src/main/java/frc/robot/control/ManualInputInterfaces.java

```diff
@@ -252,6 +252,28 @@ private void bindCommandsToDriverXboxButtons(){
               "increment power factor")
             )
           );
+        // right trigger press will put drivetrain in immoveable stance  
+        this.driverController.rightTrigger().onTrue(
+          new ParallelCommandGroup(
+            new InstantCommand(
+              () -> subsystemCollection.getDriveTrainSubsystem().setSwerveDriveMode(SwerveDriveMode.IMMOVABLE_STANCE)
+            ),
+            new ButtonPressCommand(
+            "driverController.rightTrigger()",
+            "immoveable stance")
+          )
+        );
+        // right trigger de-press will put drivetrain in normal drive mode  
+        this.driverController.rightTrigger().onFalse(
```

Comment:
I'm a smidge concerned about a case where the signal is lost in translation (controller -> PC -> Driver Station -> FMS -> Radio -> RoboRio).  If this stack hiccups, the robot will stay in X mode until a press-release happens that is received.

A couple of options:
1. We just test with it and gain confidence
2. We tell John that if X is stuck he needs to trigger on-off again until it is unstuck
3. We bake a command that reverts to NORMAL_DRIVING automatically
4. We attempt to use Trigger whileTrue/whileFalse (https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/button/Trigger.html)

Reply:
Another idea is to incorporate the revert logic (back to NORMAL_DRIVING) in the All stop command. ... Then we tell John to click the X button (slightly confusing) if the robot won't get out of X-mode.

Reply:
I like the idea of all stop command revert to NORMAL_DRIVING.  I'll implement that and update the PR.

Reply:
added this in the latest rev.